File: 455. Assign Cookies.cpp
#include"LeetCodeLibs.h"

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        int res = 0, n = g.size();
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());

        for(int& i : s){
            if(res < n && g[res] <= i) ++res;
        }

        return res;
    }
};

File: 455. Assign Cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        int res = 0, n = g.length;
        Arrays.sort(g);
        Arrays.sort(s);

        for(int i : s){
            if(res < n && g[res] <= i) ++res;
        }

        return res;
    }
}